src/main/java/bob/Bob.java

```diff
@@ -0,0 +1,237 @@
+package bob;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Scanner;
+import java.util.stream.Collectors;
+
+public class Bob {
+    public static void main(String[] args) {
+
+        final String FILENAME = "bob.txt";
+
+        displayWelcomeMessage();
+        List<Task> list = generateListOnStartup(FILENAME);
+
+        Scanner in = new Scanner(System.in);
+
+        while (true) {
+
+            String line = in.nextLine();
+            String command = line.split(" ")[0];
+
+            try {
+                boolean endLoop = processCommand(command, line, list, FILENAME);
+
+                if (endLoop) {
+                    break;
+                }
+            } catch (BobException e) {
+                displayHorizontalLine();
+                System.out.println(e.getMessage());
+                displayHorizontalLine();
+            }
+
+        }
+    }
+
+    private static List<Task> generateListOnStartup(String filename) {
+        List<String> stringList;
+        try {
+            stringList = Files.readAllLines(Paths.get(filename));
+        } catch (IOException e) {
+            // if there is any problem with the file or the file does not exist, return an empty list
+            // this means if the file is corrupted it will not be used, and overwritten later
+            return new ArrayList<>();
+        }
+
+        List<Task> list = new ArrayList<>();
+        for (String s : stringList) {
+            switch (s.substring(0,3)) {
+            case "[T]":
+                list.add(new Task(s));
+                break;
+            case "[D]":
+                list.add(new Deadline(s));
+                break;
+            case "[E]":
+                list.add(new Event(s));
```

Comment:
avoid using magic literals by storing them in static final variables 
https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers